### ISSUE TITLE
Hi~ I want to collect consul's changed history to watch what kvs were updated, and store the history into log engine. How can I reach this goal?

### ISSUE THREAD

**Issue by jianzzz** [hashicorp/consul#4804]
Issues on GitHub are intended to be related to bugs or feature requests, so we recommend using our other community resources instead of asking here if you have a question.

- Frequently asked questions: https://www.consul.io/docs/faq.html
- Consul Guides: https://www.consul.io/docs/guides/index.html

Any other questions can be sent to the Consul mailing list: https://www.consul.io/community.html

**Comment by jianzzz** [comment]
Hi~ I want to collect consul's changed history to watch what kvs were updated, and store the history into log engine. How can I reach this goal?

**Comment by pierresouchay** [comment]
Yes using watches, but you won't be sure not to miss some updates, for instance, for a given key key1, if value get updated to 1 then 2 and very quickly 3, you might miss the update 2 (unlikely, but very possible)
